api/fma/v1alpha1/launcherpoolpolicy_types.go

```diff
@@ -0,0 +1,214 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// LauncherPoolPolicy defines the proactive provisioning policy for the idle launcher pods.
+// The LauncherPoolPolicy semantics should be defined as below:
+//
+// ## Multiple LauncherPoolPolicy Objects
+//
+// 1. **Additive Semantics**
+//   - Multiple LauncherPoolPolicy objects follow additive semantics
+//   - All policies across all objects are evaluated together
+//   - Rules from different objects are combined to form the complete policy set
+//
+// 2. **Zero Objects Behavior**
+//   - When no LauncherPoolPolicy objects exist, no proactive provisioning occurs
+//   - System falls back to on-demand launcher pod creation
+//
+// ## Multiple LauncherPoolForNodeType Matching
+//
+// 1. **Multiple Matches**
+//   - When multiple LauncherPoolForNodeType structs match a single Node (across same or different LauncherPoolPolicy objects):
+//     - For each unique combination of (Node, AcceleratorSet, LauncherConfig), select the highest LauncherCount value
+//     - This forms the effective policy for that specific tuple
+//
+// 2. **Zero Matches**
+//   - When no LauncherPoolForNodeType matches a Node:
+//     - No pre-provisioning policy applies to that Node
+//     - Launcher pods for that Node are created on-demand
+//
+// ## Multiple CountForLauncher with Same LauncherConfig
+//
+// 1. **Duplicate LauncherConfig Names**
+//   - When multiple CountForLauncher structs specify the same LauncherConfigName for the same (Node, AcceleratorSet) combination:
+//     - Select the highest LauncherCount value among all matching entries
+//     - This determines the target pool size for that LauncherConfig on that Node with that Accelerator
+//
+// 2. **Zero Matching CountForLauncher**
+//   - When no CountForLauncher matches a specific (Node, AcceleratorSet, LauncherConfig) tuple:
+//     - No pre-provisioning occurs for that specific combination
+//     - Launcher pods are created on-demand when needed
+//
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:shortName=lpp
+
+type LauncherPoolPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   LauncherPoolPolicySpec   `json:"spec,omitempty"`
+	Status LauncherPoolPolicyStatus `json:"status,omitempty"`
+}
+
+// LauncherPoolPolicySpec defines the node-level idle pool configuration.
+type LauncherPoolPolicySpec struct {
+	// LauncherPoolForNodeType defines pool spec per node type.
+	LauncherPoolForNodeType []LauncherPoolForNodeType `json:"launcherPoolForNodeType"`
```

Comment:
The current design has two levels of grouping, and I do not understand why that is needed. There can be multiple LauncherPoolPolicy objects, and each does nothing but bundle together multiple LauncherPoolForNodeType structs.